### PR TITLE
New Validation Correlation rules - Missing search window

### PR DESCRIPTION
**Pull request by michal-dagan**
## Related Issues
fixes:https://jira-hq.paloaltonetworks.local/browse/CIAC-7254

## Description
Added a **validate** check in correlation rule files making sure that if `execution_mode` is set to `SCHEDULED`, `search_window` cannot be empty.
On the other hand, when `execution_mode` is set to `REAL_TIME`, `search_window` can be empty.

### CODE REVIEW

File: demisto_sdk/commands/common/hook_validations/correlation_rule.py [hunk 1]
```diff
@@ -84,3 +85,16 @@ def is_files_naming_correct(self):
                 self._is_valid = False
                 return False
         return True
+
+    @error_codes("CR102")
+    def validate_execution_mode_search_window(self):
+        """
+        Validates 'search_window' existence and non-emptiness for 'execution_mode' = 'SCHEDULED'.
+        """
+        if (not 'search_window' in self.current_file or
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/correlation_rule.py:94]:
```suggestion
        if (('search_window' not in self.current_file) or
```
unless it's not what you meant

Comment (michal-dagan) [reply demisto_sdk/commands/common/hook_validations/correlation_rule.py]: it is still a draft PR

File: demisto_sdk/commands/common/errors.py [hunk 2]
```diff
@@ -4373,3 +4377,8 @@ def hidden_pack_not_mandatory_dependency(
     @error_code_decorator
     def pack_have_nonignorable_error(nonignorable_errors: List[str]):
         return f"The following errors can not be ignored: {', '.join(nonignorable_errors)}, remove them from .pack-ignore files"
+
+    @staticmethod
+    @error_code_decorator
+    def correlation_rules_execution_mode_error():
+        return "The 'search_window' key must exist and cannot be empty when the 'execution_mode' is set to 'SCHEDULED'."
```
Comment (bziser) [inline demisto_sdk/commands/common/errors.py:4384]:
```suggestion
        return "When 'execution_mode' is set to 'SCHEDULED', the 'search_window' key must exist and cannot be empty."
```

File: CHANGELOG.md [hunk 3]
```diff
@@ -11,6 +11,7 @@
 * Fixed an issue where the (`GR108`) validation did not fail in the validate command with the `-a` flag.
 * Modified **prepare_content** command to be platform specific. For xsoar-saas and XSIAM regarding pack readme and integration description images in markdown files.
 * Fixed an issue where the **lint** command was parsing % that may exist in the log data.
+* Added a **validate** check in correlation rule files making sure that if `execution_mode` is set to `SCHEDULED`, `search_window` cannot be empty.
```
Comment (dorschw) [inline CHANGELOG.md:14]:
```suggestion
* Added a **validate** check for correlation rule files, making sure that `search_window` cannot be empty when `execution_mode` is set to `SCHEDULED`.
```

File: demisto_sdk/commands/common/errors.py [hunk 4]
```diff
@@ -1404,6 +1404,10 @@
         "code": "CR101",
         "related_field": "",
     },
+    "correlation_rules_execution_mode_error": {
```
Comment (dorschw) [inline demisto_sdk/commands/common/errors.py:1407]:
```suggestion
    "correlation_rules_execution_mode_empty: {
```
gives a tiny bit more info, the word `error` is redundant

File: demisto_sdk/commands/common/hook_validations/correlation_rule.py [hunk 5]
```diff
@@ -84,3 +85,18 @@ def is_files_naming_correct(self):
...
+        """
+        Validates 'search_window' existence and non-emptiness for 'execution_mode' = 'SCHEDULED'.
+        """
+        if ("search_window" not in self.current_file) or (
+            self.current_file["execution_mode"] == "SCHEDULED"
+            and not self.current_file["search_window"]
+        ):
+            error_message, error_code = Errors.correlation_rules_execution_mode_error()
+            if self.handle_error(error_message, error_code, file_path=self.file_path):
+                self._is_valid = False
```
Comment (dorschw) [inline demisto_sdk/commands/common/hook_validations/correlation_rule.py:100]: do we usually change `_is_valid`? Shouldn't it be modified by whatever method calls this one?

**Review by dorschw — APPROVED** [review]

**Review by bziser — APPROVED** [review]

**Pull request merged**